cpp/src/arrow/compute/exec.cc

```diff
@@ -1049,10 +1057,22 @@ class VectorExecutor : public KernelExecutorImpl<VectorKernel> {
         RETURN_NOT_OK(Exec(span, listener));
       }
     } else {
-      // Kernel cannot execute chunkwise. If we have any chunked
-      // arrays, then VectorKernel::exec_chunked must be defined
-      // otherwise we raise an error
+      // Some vector kernels have a separate code path for handling
+      // chunked arrays (VectorKernel::exec_chunked), so we check if we
+      // have any chunked arrays. If we do and an exec_chunked function
+      // is defined then we call that.
+      bool have_chunked_arrays = false;
```

Comment:
`have_chunked_arrays` only used in `kernel_->can_execute_chunkwise=false`'s branch.

Reply:
I like this code move. That i could only understand when looking commit by commit.